tls/_common/test/test_constructs.py

```diff
@@ -336,6 +337,57 @@ def test_round_trip(self, tls_array, ints, uint8_encoded):
         assert tls_array.parse(unparsed) == ints
 
 
+@pytest.mark.parametrize(
+    "ints,uint8_encoded",
+    [([], b'\x00\x00\x00' + b''),
+     ([1, 2, 3], b'\x00\x00\x03' + b'\x01\x02\x03'),
+     ([1] * 65535, b'\x00\xFF\xFF' + b'\x01' * 65535)])
+class TestTLSPrefixedArrayWithCustomLengthFieldSize(object):
+    """
+    Tests for :py:func:`tls._common._constructs.TLSPrefixedArray` where the
+    length_field is supplied by the user.
+    """
+
+    @pytest.fixture
+    def tls_array(self):
+        """
+        A :py:func:`tls._common._constructs.TLSPrefixedArray` of
+        :py:func:`construct.macros.UBInt8`.
+        """
+        return TLSPrefixedArray("digits", UBInt8("digit"),
+                                length_field_size=UBInt24)
+
+    def test_build(self, tls_array, ints, uint8_encoded):
+        """
+        A :py:meth:`tls._common._constructs.TLSPrefixedArray` specialized on a
+        given :py:func:`construct.Construct` encodes a sequence of objects as a
```

Comment:
Maybe mention that the length is 24 bits because the default 16 bit length has been overridden?